File: B13/Array/Hemanth/Move_zeroes.java
//Leetcode link : https://leetcode.com/problems/move-zeroes/

//Steps :
//1. Count no of zeroes in an array.
//2. Place all the non-zero elements into the array.
//3. Then place all the zeroes at the end of an array

class Solution {
    public void moveZeroes(int[] nums) {
        int zeros=0;
        int i=0;
        for(int it : nums){
            if(it==0){
                zeros++;
            }else{
                nums[i++]=it;
            }
        }
        while(zeros>0){
            nums[i++]=0;
            zeros--;
        }
    }
}

//Time Complexity : O(n)
//Space Complexity : O(1)

File: B13/Subhendu/Array/move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        ## https://leetcode.com/problems/move-zeroes/
        idx = 0
        zero = 0
        for i in range(len(nums)):
            if nums[i] != 0:
                nums[idx] = nums[i]
                idx += 1
            else:
                zero += 1

        while zero:
            nums[idx] = 0
            idx += 1
            zero -= 1

        return nums